src/Std/Internal/Async/DNS.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time
+import Std.Internal.UV
+import Std.Internal.Async.Basic
+import Std.Net.Addr
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+namespace DNS
+
+open Std.Net
+
+/--
+This type specifies the desired address family for the returned addresses in `getAddrInfo`.
+-/
+inductive AddressFamily
+
+  /--
+  IPv4 address family.
+  -/
+  | ipv4
+
+  /--
+  IPv6 address family.
+  -/
+  | ipv6
+deriving Repr, DecidableEq
```

Comment:
We already have this type here https://github.com/leanprover/lean4/blob/master/src/Std/Net/Addr.lean#L85-L91